### PR TITLE
executor: fix load data deal with escape character.

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve?

According to mysql document,  the escaped character is interpreted as if it was not escaped. For example, \x is just x. When TiDB load data  interpreting escaped, it ignore following character, just deal with character like `\t`.

### What is changed and how it works?

If the escaped character is interpreted as if it was not escaped, output the original character. For example:

`9002	tab\	tab`

will be interpreted as:

`9002	tab	tab`

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- No

### CODE REVIEW

File: executor/load_data.go [hunk 1]
```diff
@@ -632,14 +632,10 @@ func (w *fieldWriter) GetField() (bool, field) {
...
 			// TODO: escape only support '\'
-			w.OutputBuf = append(w.OutputBuf, ch)
 			flag, ch = w.getChar()
 			if flag {
-				if ch == w.enclosedChar {
-					w.OutputBuf = append(w.OutputBuf, ch)
-				} else {
-					w.putback()
-				}
+				w.OutputBuf = append(w.OutputBuf, '\\')
```
Comment (cfzjywxk) [inline executor/load_data.go:637]: Better to add some comments to explain this here

Comment (imtbkcat) [reply executor/load_data.go:639]: added.

Comment (cfzjywxk) [reply executor/load_data.go:639]: If the backslash is followed by another backslash, like `"9\ttab\\\\	tab\n"`, will it return `"9\ttab\\	tab\n"`  incorrectly ?

Comment (imtbkcat) [reply executor/load_data.go:639]: @cfzjywxk it will return `"9	tab\\	tab"` since `\t` also mean a tab

**Comment by imtbkcat** [comment]
PTAL @jackysp @cfzjywxk

**Comment by cfzjywxk** [comment]
LGTM

**Pull request merged**